src/supernova/mod.rs

```diff
@@ -810,3 +812,14 @@ where
 
   (ck_primary, ck_secondary)
 }
+
+/// SuperNova helper trait, for implementors that provide sets of sub-circuits to be proved via NIVC.
+pub trait CircuitSet<G: Group> {
```

Comment:
Are you looking for an heterogeneous list (or more likely ` ExactSizeIterator`) of `dyn StepCircuit<F>`, where `F: Primefield`?

Reply:
Maybe, but this trait will end up doing some heavy lifting, and it's not really clear yet what it needs to look like. The above is just a first foot in the door. I will iterate in future revisions and bear your ideas in mind.